packages/nocodb/src/lib/version-upgrader/NcUpgrader.ts

```diff
@@ -27,96 +27,109 @@ export default class NcUpgrader {
     this.log(`upgrade :`);
     let oldVersion;
 
-    try {
-      ctx.ncMeta = await ctx.ncMeta.startTransaction();
-
-      const NC_VERSIONS: any[] = [
-        { name: '0009000', handler: null },
-        { name: '0009044', handler: null },
-        { name: '0011043', handler: ncProjectEnvUpgrader },
-        { name: '0011045', handler: ncProjectEnvUpgrader0011045 },
-        { name: '0090000', handler: ncProjectUpgraderV2_0090000 },
-        { name: '0098004', handler: ncDataTypesUpgrader },
-        { name: '0098005', handler: ncProjectRolesUpgrader },
-        { name: '0100002', handler: ncFilterUpgrader },
-        { name: '0101002', handler: ncAttachmentUpgrader },
-        { name: '0104002', handler: ncAttachmentUpgrader_0104002 },
-      ];
-      if (!(await ctx.ncMeta.knexConnection?.schema?.hasTable?.('nc_store'))) {
-        return;
-      }
-      this.log(`upgrade : Getting configuration from meta database`);
+    const NC_VERSIONS: any[] = [
+      { name: '0009000', handler: null },
+      { name: '0009044', handler: null },
+      { name: '0011043', handler: ncProjectEnvUpgrader },
+      { name: '0011045', handler: ncProjectEnvUpgrader0011045 },
+      { name: '0090000', handler: ncProjectUpgraderV2_0090000 },
+      { name: '0098004', handler: ncDataTypesUpgrader },
+      { name: '0098005', handler: ncProjectRolesUpgrader },
+      { name: '0100002', handler: ncFilterUpgrader },
+      { name: '0101002', handler: ncAttachmentUpgrader },
+      { name: '0104002', handler: ncAttachmentUpgrader_0104002 },
+    ];
+    if (!(await ctx.ncMeta.knexConnection?.schema?.hasTable?.('nc_store'))) {
+      return;
+    }
+    this.log(`upgrade : Getting configuration from meta database`);
+
+    const config = await ctx.ncMeta.metaGet('', '', 'nc_store', {
+      key: this.STORE_KEY,
+    });
+
+    if (config) {
+      const configObj: NcConfig = JSON.parse(config.value);
+      if (configObj.version !== process.env.NC_VERSION) {
+        oldVersion = configObj.version;
+        let fromVersion = configObj.version;
```

Comment:
The naming for `oldVersion` and `fromVersion` is a bit confusing. At the first sight, I thought they are referring to the same thing. What about 

```
oldVersion -> previousVersion
fromVersion -> currentVersion
```

and we can set 

```
latestVersion: process.env.NC_VERSION
```

with some comments like

- `previousVersion` is the version before starting the upgrader
- `currentVersion` is the current version being upgraded to
- `latestVersion` is the latest / target / final version to be upgraded to

What do you think? Or you may forget about the rename part but I believe comments are necessary here.